device/transport/amqp/src/amqp_twin_client.ts

```diff
@@ -6,102 +6,134 @@
 import { EventEmitter } from 'events';
 import machina = require('machina');
 
-import { endpoint, errors, results, Message, AuthenticationProvider } from 'azure-iot-common';
-import { Amqp as BaseAmqpClient, translateError, AmqpMessage } from 'azure-iot-amqp-base';
+import { endpoint, Message, AuthenticationProvider } from 'azure-iot-common';
+import { Amqp as BaseAmqpClient, AmqpMessage, SenderLink, ReceiverLink } from 'azure-iot-amqp-base';
+import { TwinProperties } from 'azure-iot-device';
 
 import * as uuid from 'uuid';
 import * as dbg from 'debug';
 const debug = dbg('azure-iot-device-amqp:AmqpTwinClient');
 
-const responseTopic = '$iothub/twin/res';
+enum TwinMethod {
+  GET = 'GET',
+  PATCH = 'PATCH',
+  PUT = 'PUT',
+  DELETE = 'DELETE'
+}
+
 
 /**
  * @private
  * @class        module:azure-iot-device-amqp.AmqpTwinClient
- * @classdesc    Acts as a receiver for device-twin traffic
+ * @classdesc    Acts as a client for device-twin traffic
  *
- * @param {Object} config   configuration object
- * @fires AmqpTwinClient#subscribed   an response or post event has been set up for listening.
- * @fires AmqpTwinClient#error    an error has occurred
- * @fires AmqpTwinClient#response   a response message has been received from the service
- * @fires AmqpTwinClient#post a post message has been received from the service
- * @throws {ReferenceError} If client parameter is falsy.
+ * @param {Object} config                        configuration object
+ * @fires AmqpTwinClient#error                   an error has occurred
+ * @fires AmqpTwinClient#desiredPropertyUpdate   a desired property has been updated
+ * @throws {ReferenceError}                      If client parameter is falsy.
  *
  */
 
 /* Codes_SRS_NODE_DEVICE_AMQP_TWIN_06_005: [The `AmqpTwinClient` shall inherit from the `EventEmitter` class.] */
 export class AmqpTwinClient extends EventEmitter {
-  static errorEvent: string = 'error';
-  static responseEvent: string = 'response';
-  static postEvent: string = 'post';
-  static subscribedEvent: string = 'subscribed';
-
   private _client: BaseAmqpClient;
   private _authenticationProvider: AuthenticationProvider;
-  private _boundMessageHandler: Function;
   private _endpoint: string;
-  private _upstreamAmqpLink: any;
-  private _downstreamAmqpLink: any;
+  private _senderLink: SenderLink;
+  private _receiverLink: ReceiverLink;
   private _fsm: any;
-  private _internalOperations: { [key: string]: () => void };
-  private _eventQueue: any[];
-  private _eventQueueError: Error;
+  // need type for twin properties
```

Comment:
Mark with TODO: if this is a todo.  Otherwise, it looks like you're trying to describe what the code does